apps/explorer/src/ui/Button.tsx

```diff
@@ -35,10 +36,33 @@ const buttonStyles = cva(
 
 export interface ButtonProps
     extends VariantProps<typeof buttonStyles>,
-        ButtonOrLinkProps {}
+        ButtonOrLinkProps {
+    loading?: boolean;
+}
 
-export function Button({ variant, size, ...props }: ButtonProps) {
+export function Button({
+    variant,
+    size,
+    loading,
+    children,
+    ...props
+}: ButtonProps) {
     return (
-        <ButtonOrLink className={buttonStyles({ variant, size })} {...props} />
+        <ButtonOrLink
+            className={buttonStyles({ variant, size })}
+            {...props}
+            disabled={props.disabled || loading}
+        >
+            {loading ? (
+                <>
+                    <div className="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2">
+                        <LoadingSpinner />
+                    </div>
+                    <div className="invisible">{children}</div>
```

Comment:
I think this makes it invisible to screen readers, maybe `text-transparent`?

Reply:
Done